test/test_torch.py

```diff
@@ -6521,41 +6522,61 @@ def test_logical_not(self, device):
             a.logical_not_()
             self.assertEqual(expected_res, a)
 
-    def test_logical_xor(self, device):
+    def _test_logical(self, device, op, a_, b_, expected_res_):
         for dtype in torch.testing.get_all_dtypes():
-            expected_res = torch.tensor([0, 0, 1, 1], dtype=dtype, device=device)
-            a = torch.tensor([10, 0, 1, 0], dtype=dtype, device=device)
+            if device.startswith('xla') and dtype == torch.half:  # Skip Half on xla.
```

Comment:
Hmmm we shouldn't put 'xla' in the code branches. This is very hard to maintain... It's okay for XLA for fail on Half for now. (Let's skip in pytorch/xla). 
We'll have a more generalized way to let XLA skip half tests in the future.

Reply:
OK, I have removed the relevant branches. Could you merge https://github.com/pytorch/xla/pull/1325 then?

Reply:
Thanks! Rebased -- expecting all tests to pass